### ISSUE TITLE
URI-Routing

### ISSUE THREAD

**Issue by ignoreigor** [tine20/tine20#7383]
Hi,

as I mentioned after the last nightly release https://github.com/tine20/tine20/commit/43179e24c44e0231125a7109c27f77b263dee9b5 (maybe stupid position for a comment) the fix for routing disabled my routing. While reverting changes in FastRoute.php in release 2022.06.11.32, I did some debugging with 2022.06.18.130.

My problem is caused by the preg_replace in Line 52. Subdir-path e.g. /tinesubdir/logo/i is replaced to //logo/i, which is not found.
For me replacing the subdir is better than changing it to '/' 

`$uri = preg_replace('/^' . preg_quote($serverPath, '/') . '/', '', $uri);`

results in /logo/i - which is found.

tine20URL is set to
    'tine20URL' => 'https://tineserver.de/tinesubdir',

Maybe you'll have another look on that.

**Comment by pschuele** [comment]
maybe we can just remove duplicate "/" in the uri?

i added another preg_replace after the $serverPath if:

```php
        // if the tine20 server is located in a subdir, we need to remove the server path from the uri
        if ('/' !== ($serverPath = Tinebase_Core::getUrl(Tinebase_Core::GET_URL_PATH)) && '' !== $serverPath) {
            $uri = preg_replace('/^' . preg_quote($serverPath, '/') . '/', '/', $uri);
        }

        // remove multiple slashes as well - FastRoute is not handling uris with them correctly it seems
        $uri = preg_replace('/\/+/', '/', $uri);
```

can you please give it a try?

**Comment by ignoreigor** [comment]
Hi Philipp,

> can you please give it a try?

thanks for your effort. The behavior is still visible in version 2022.08.13.154, as the code at this point has not been changed.

Your replacement works.  Wouldn't it be safer to start the replacement with ^ to replace only at the beginning? Presumably // should not occur anywhere.

Best regards.

**Comment by ignoreigor** [comment]
Fixed in 2022.08.20.85. Thank you!
